db/migrate/20190116182055_add_discovery_tables.rb

```diff
@@ -112,50 +112,54 @@ def up
           CONSTRAINT commits_author_vcs_user_id_fkey FOREIGN KEY (author_vcs_user_id) REFERENCES vcs_users(id),
           CONSTRAINT commits_committer_vcs_user_id_fkey FOREIGN KEY (committer_vcs_user_id) REFERENCES vcs_users(id)
         );
-        
+
         CREATE
             TRIGGER trigger_set_timestamp BEFORE INSERT
                 OR UPDATE
                     ON
                     commits FOR EACH ROW EXECUTE PROCEDURE trigger_set_timestamp();
-        
+
         CREATE TABLE IF NOT EXISTS branches (
           id serial NOT NULL,
           branch_name text NOT NULL,
           branch_type branch_type NULL,
           repo_id int4 NULL,
+          project_id int4 NULL,
           created_at timestamp NOT NULL,
           updated_at timestamp NOT NULL,
           CONSTRAINT branches_pkey PRIMARY KEY (id),
-          CONSTRAINT branches_repo_id_fkey FOREIGN KEY (repo_id) REFERENCES repos(id)
+          CONSTRAINT branches_repo_id_fkey FOREIGN KEY (repo_id) REFERENCES repos(id) ON DELETE CASCADE,
+          CONSTRAINT branches_project_id_fkey FOREIGN KEY (project_id) REFERENCES projects(id) ON DELETE CASCADE
         );
         CREATE UNIQUE INDEX IF NOT EXISTS branches_branch_name_repo_id_idx ON branches USING btree (branch_name, repo_id);
-        
+
         CREATE
             TRIGGER trigger_set_timestamp BEFORE INSERT
                 OR UPDATE
                     ON
                     branches FOR EACH ROW EXECUTE PROCEDURE trigger_set_timestamp();
-        
+
         CREATE TABLE IF NOT EXISTS nodes (
           id serial NOT NULL,
           type node_type NOT NULL,
           id_hash text NOT NULL,
           visibility node_visibility NOT NULL,
           repo_id int4 NOT NULL,
+          project_id int4 NULL,
```

Comment:
We need to update API and Analyzer to set project_id when calling the /analyzer.process endpoint now. Is there a story for that?

Reply:
So I created this https://stoplightio.atlassian.net/browse/SL-1429, meant as a catch all. This was something I a caught.

Reply:
ok awesome! As long as we don't forget to do that work at some point 😄 .